Library/Homebrew/cmd/doctor.rb

```diff
@@ -424,10 +424,18 @@ def check_access_share_man
     __check_subdir_access "share/man"
   end
 
+  def check_access_homebrew_repository
+    unless HOMEBREW_REPOSITORY.writable_real? then <<-EOS.undent
+      The #{HOMEBREW_REPOSITORY} is not writable.
+      You should probably `chown` #{HOMEBREW_REPOSITORY}
+    EOS
+    end
+  end
+
   def check_access_usr_local
     return unless HOMEBREW_PREFIX.to_s == "/usr/local"
 
-    unless File.writable_real?("/usr/local") then <<-EOS.undent
+    unless HOMEBREW_PREFIX.writable_real? then <<-EOS.undent
```

Comment:
While this method only checks when HOMEBREW_PREFIX == /usr/local. so it's the same.